src/app/lib/select/select.component.ts

```diff
@@ -1,22 +1,56 @@
-import { Component, Input, OnInit } from '@angular/core';
+import { Component, Input, OnInit, forwardRef } from '@angular/core';
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
 
 @Component({
   selector: 'hc-select',
   templateUrl: './select.component.html',
-  styleUrls: ['./select.component.scss']
+  styleUrls: ['./select.component.scss'],
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => SelectComponent),
+      multi: true
+    }
+  ]
 })
-export class SelectComponent implements OnInit {
+export class SelectComponent implements ControlValueAccessor {
 
   @Input() placeholder: string;
-  @Input() options = [];
+  @Input() options: Array<string> = [];
   @Input() disabled: boolean = false;
-  alpha = 1.0;
+  disabledAlpha = 0.4;
+  enabledAlpha = 1.0;
+  _value: string;
+  onChange: any = () => { };
+  onTouched: any = () => { };
+
+  get alpha() {
+    return this.disabled ? this.disabledAlpha : this.enabledAlpha;
+  }
 
-  constructor() { }
+  get value() {
+    return this._value;
+  }
+
+  set value(val: string) {
+    this._value = val;
+    this.onChange(val);
+    this.onTouched();
+  }
+
+  constructor() {}
+
+  registerOnChange(fn: any) {
+    this.onChange = fn;
+  }
+
+  registerOnTouched(fn: any) {
+    this.onTouched = fn;
+  }
 
-  ngOnInit() {
-    if (this.disabled) {
-      this.alpha = 0.4;
+  writeValue(value: string) {
```

Comment:
Im trying to figure out what this is for?

Reply:
`writeValue` takes a value from a reactive form and sets it in the component. So if we created a formGroup in our host component and patched the values from an ajax call for example, the select element's value would be replaced with the new form value.